e2e/app/component/dialog.ts

```diff
@@ -41,17 +36,17 @@ export default class Dialog {
   }
 
   async waitUntilDialogIsClosed() {
-    await this.page.waitForSelector(SELECTOR.dialogRoot, {visible: false, timeout: 60000});
+    await this.page.waitForXPath(this.xpath, {visible: false, timeout: 60000});
   }
 
   private async findDialog() {
     if (this.dialogElement === undefined) {
-      this.dialogElement = await this.page.waitForSelector(SELECTOR.dialogRoot, {visible: true});
+      this.dialogElement = await this.page.waitForXPath(this.xpath, {visible: true});
     }
   }
 
   private getButtonSelector() {
-    return `${SELECTOR.dialogRoot} ${SELECTOR.button}`;
+    return `${this.xpath}//button | ${this.xpath}//*[@role="button"]`;
```

Comment:
There's some redundancy here because the `{this.xpath}` is repeated. I'd concatenate the string "${this.xpath}" with either side or else move the conditional into the descendant (after the xpath), if that's possible. There's probably also a way to join path segments richly.

Reply:
There is a buttonXpath function in xpathDefaults, could that be used and/or modified to accept a container. That would help to ensure that we are always looking for buttons in the same way